### DISCUSSION TITLE
Feature Request: I want to make the validation rules conditional on button clicks.

### DISCUSSION THREAD

**Discussion by toki-developer** [react-hook-form/react-hook-form#12182] [Q&A]
## **Is your feature request related to a problem? Please describe.**

I want to switch validation when there are multiple buttons on one form.
For example, if I have "Save Draft" and "Finalize", I want to make only the "Finalize" one "Required".

## **Describe the solution you'd like**

Allow multiple validation resolvers to be registered, so that the validation can be selected and executed by a button clicked.

## **Describe alternatives you've considered**

Functionality to register validations with id in useForm and switch between validations by passing the id of the validation as an argument to the handleSubmit function.

**Comment by bluebill1049** [comment]
you can get the `handleSubmit` i believe second argument with event target of the button value
